src/apps/allbridge/shortcuts.ts

```diff
@@ -74,14 +77,28 @@ const hook: ShortcutsHook = {
               functionName: 'deposit',
               args: [amountToSupply],
             }),
-            // TODO: consider moving this concern to the runtime
-            // which would simulate the transaction(s) to get these
-            gas: GAS,
-            estimatedGasUse: GAS / 3n,
           }
 
           transactions.push(supplyTx)
 
+          // TODO: consider moving this concern to the runtime
+          try {
+            const simulatedTransactions = await simulateTransactions({
+              transactions,
+              networkId,
+            })
+            const supplySimulatedTx =
+              simulatedTransactions[simulatedTransactions.length - 1]
+
+            supplyTx.gas =
+              BigInt(supplySimulatedTx.gasNeeded) +
+              SIMULATED_DEPOSIT_GAS_PADDING
+            supplyTx.estimatedGasUse = BigInt(supplySimulatedTx.gasUsed)
+          } catch (error) {
+            supplyTx.gas = GAS
```

Comment:
do we need a log here? if this is due to an unexpected error, seems like it would be useful for debugging

Reply:
yeah, probably.  the annoying thing is that there will always be an error on Celo (because no simulation support).